src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -45,7 +45,6 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
                 case BoundKind.ConditionalReceiver:
                     // do nothing receiver ref must be already pushed
                     Debug.Assert(!expression.Type.IsReferenceType);
-                    Debug.Assert(!expression.Type.IsValueType);
```

Comment:
Consider relaxing this assert for Nullable types rather than removing it altogether.